From: SeongJae Park <sj@kernel.org>
Date: Sat, 6 Sep 2025 19:26:29 -0700
Subject: [PATCH] mm/damon/core: use damos_commit_quota_goal() for new goal
 commit

When a new quota goal is added via damos_commit_quota_goals(), the new
quota sets the metric and target value using damos_new_quota_goal(), and
set per-metric optional fields via damos_commit_quota_goal_union().
However, it is missing setting the current_value field, which should be
set if the metric is DAMOS_QUOTA_USER_INPUT.  Fix this by using
damos_commit_quota_goal() instead of damos_commit_quota_goal_union().

Fixes: 1aef9df0ee90 ("mm/damon/core: commit damos_quota_goal->nid")
Cc: <stable@vger.kernel.org> # 6.16.x
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 93848b4c6944..e72dc49d501c 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -832,7 +832,7 @@ int damos_commit_quota_goals(struct damos_quota *dst, struct damos_quota *src)
 				src_goal->metric, src_goal->target_value);
 		if (!new_goal)
 			return -ENOMEM;
-		damos_commit_quota_goal_union(new_goal, src_goal);
+		damos_commit_quota_goal(new_goal, src_goal);
 		damos_add_quota_goal(dst, new_goal);
 	}
 	return 0;
-- 
2.47.3

